// @bit/akameco.styled-spinkit.internal.util/dist/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_components_1 = require("styled-components");
exports.size = function (width, height) {
    if (height === void 0) { height = width; }
    return styled_components_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  width: ", ";\n  height: ", ";\n"], ["\n  width: ", ";\n  height: ", ";\n"])), width, height);
};
exports.sizePx = function (n) { return exports.size(n + "px"); };
// eslint-disable-next-line no-shadow
exports.propSize = function (_a) {
    var size = _a.size;
    return exports.sizePx(size);
};
exports.propBgColor = function (_a) {
    var color = _a.color;
    return styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  background-color: ", ";\n"], ["\n  background-color: ", ";\n"])), color);
};
// eslint-disable-next-line no-shadow
exports.propMargin = function (_a) {
    var _b = _a.size, size = _b === void 0 ? 0 : _b;
    return styled_components_1.css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  margin: ", "px auto;\n"], ["\n  margin: ", "px auto;\n"])), size);
};
exports.animationDelay = function (n) {
    return styled_components_1.css(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n    animation-delay: ", "s;\n  "], ["\n    animation-delay: ", "s;\n  "])), n);
};
exports.propDelay = function (_a) {
    var delay = _a.delay;
    return exports.animationDelay(delay);
};
exports.getRange = function (n) {
    return Array.from(new Array(n).keys());
};
var templateObject_1, templateObject_2, templateObject_3, templateObject_4;
